v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/SpannerToSourceDb.java

```diff
@@ -438,4 +594,27 @@ public static SpannerIO.ReadChangeStream getReadChangeStreamDoFn(
     }
     return readChangeStreamDoFn;
   }
+
+  private static DeadLetterQueueManager buildDlqManager(Options options) {
+    String tempLocation =
+        options.as(DataflowPipelineOptions.class).getTempLocation().endsWith("/")
+            ? options.as(DataflowPipelineOptions.class).getTempLocation()
+            : options.as(DataflowPipelineOptions.class).getTempLocation() + "/";
+    String dlqDirectory =
+        options.getDeadLetterQueueDirectory().isEmpty()
+            ? tempLocation + "dlq/"
+            : options.getDeadLetterQueueDirectory();
+    LOG.info("Dead-letter queue directory: {}", dlqDirectory);
+    options.setDeadLetterQueueDirectory(dlqDirectory);
+    if ("regular".equals(options.getRunMode())) {
+      return DeadLetterQueueManager.create(dlqDirectory, options.getDlqMaxRetryCount());
+    } else {
+      String retryDlqUri =
+          FileSystems.matchNewResource(dlqDirectory, true)
+              .resolve("severe", StandardResolveOptions.RESOLVE_DIRECTORY)
+              .toString();
+      LOG.info("Dead-letter retry directory: {}", retryDlqUri);
+      return DeadLetterQueueManager.create(dlqDirectory, retryDlqUri, 0);
```

Comment:
This makes severe directory as the retryable directory. Which implies that all the errors irrespective of retryable or severe will be retried in case of "retryDLQ" run mode. Is this intended behaviour?

Reply:
Yeah, this was intended. The retryDLQ mode is when customer will fix all the severe errors and rerun.